drivers/soundwire/intel_ace2x.c

```diff
@@ -286,6 +286,34 @@ static int intel_params_stream(struct sdw_intel *sdw,
 	return -EIO;
 }
 
+static int intel_prepare_stream(struct sdw_intel *sdw,
+				struct snd_pcm_substream *substream,
+				struct snd_soc_dai *dai,
+				struct snd_pcm_hw_params *hw_params,
+				int link_id, int alh_stream_id)
+{
+	struct sdw_intel_link_res *res = sdw->link_res;
+	struct sdw_intel_stream_params_data params_data;
+	int ret;
+
+	params_data.substream = substream;
+	params_data.dai = dai;
+	params_data.hw_params = hw_params;
+	params_data.link_id = link_id;
+	params_data.alh_stream_id = alh_stream_id;
+
+	if (res->ops && res->ops->params_stream && res->dev) {
+		ret = res->ops->params_stream(res->dev, &params_data);
+		if (ret < 0)
+			return ret;
+	}
+
+	if (res->ops && res->ops->prepare_stream && res->dev)
+		return res->ops->prepare_stream(res->dev, &params_data);
+
+	return -EIO;
+}
```

Comment:
Don't we need to do the same in drivers/soundwire/intel.c for MTL and older platforms?
BTW, I think it is better to split the commit into 2 commits. I.e. Add the prepare_stream callback first the then use it.

Reply:
No, we dont. MTL and older platforms do not usethe HD-Audio DMA

Reply:
yes, I do need to plit up this commit and reorder it to go before the actual refactoring. I will fix it in the next iteration